### PR TITLE
Refactor to typescript

### PR DESCRIPTION
**Pull request by davidmfinol**
#### Changes

- Refactor to typescript

#### Checklist

- [x] Read the contribution [guide](../CONTRIBUTING.md) and accept the [code](../CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by GabLeRoux — COMMENTED** [review]
I assume it's normal to have the `dist` folder tracked (for example, unity-builder does the same). Just asking to make sure it's normal to have that many changed lines: `+24,769 −3,191`.

**Review by GabLeRoux — APPROVED** [review]
I'm seeing some duplicated code between repositories. I'm wondering how we could solve this, but again, this should most likely be solved later. Thanks for converting this to typescript :tada: :shipit:

**Comment by davidmfinol** [comment]
Yep, I basically copy-pasted a lot of code between `unity-request-activation-file`, `unity-activate`, `unity-test-runner`, and `unity-builder`.
I agree that there should be a way to have a single repo (in `unity-actions`?) that hosts that shared code, but I think that change can be made after all the actions are already using a very similar code base.

**Review by webbertakken — CHANGES_REQUESTED** [review]
we need to upgrade to v1 of docker images and decide whether we want to keep a rolling tag (like we do now with v0) or lock it to minor level of the version.

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
+if [ -t 1 ]; then
+  exec < /dev/tty
+fi
+
 npx lint-staged
```
[the remark this answers is not in the document]
Comment (davidmfinol) [reply .husky/pre-commit]: Updated

**Comment by davidmfinol** [comment]
> we need to upgrade to v1 of docker images and decide whether we want to keep a rolling tag (like we do now with v0) or lock it to minor level of the version.

@webbertakken Moving to v1 of docker images will be done in a follow-up, right? Is it ok to merge this PR now?

**Comment by GabLeRoux** [comment]
> > we need to upgrade to v1 of docker images and decide whether we want to keep a rolling tag (like we do now with v0) or lock it to minor level of the version.
> 
> @webbertakken Moving to v1 of docker images will be done in a follow-up, right? Is it ok to merge this PR now?

I tend to say yes but I’m unsure of the implications  and would prefer @webbertakken ’s input on this

**Comment by webbertakken** [comment]
Yes, but just know that the version field in package.json will not be auto-updated (as it's not a public npm package) so setting it to v2.0.0 might be confusing.

image

On the other hand 🤷‍♂️

**Review by webbertakken — APPROVED** [review]

**Pull request merged**